File: python/005.py
"""
SMALLEST MULTIPLE
Problem 5
2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
"""
import math     # to use for built-in gcd function

def lcm(a, b):
    return abs(a*b) // math.gcd(a,b)    # finding lcm using gcd
    
def lcm_range(n):
    result = 1
    for i in range(2, n+1):
        result = lcm(result, i)     # compute running lcm
    return result
    
print(lcm_range(20))

File: javascript/005.js
/* 
SMALLEST MULTIPLE
Problem 5
2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
*/

function gcd(a,b){
    while (b !== 0){
        [a, b] = [b, a % b];    // euclidean algorithm
    }
    return a;
}

function lcm(a, b){
    return (a * b) / gcd(a, b);     // lcm using gcd
}

function lcmRange(n){
    let result = 1;
    for (let i = 2; i <= n; i++){
        result = lcm(result, i);    // update result with cumulative lcm
    }
    return result;
}

console.log(lcmRange(20));
